### ISSUE TITLE
Support for Multiple Geopoints Filter

### ISSUE THREAD

**Issue by becauseinterwebs** [typesense/typesense#958]
We have an app where the user can select multiple locations (geopoint field defined as 'coords' in our schema) to search and a radius for each geopoint.  We are trying to filter results based on that array of geopoints where the result document falls within one of the geopoints (like country:['US','UK],) however only one geopoint filter is supported.

## Steps to reproduce

```
const q = {
    q: "*",
    query_by: "name",
    sort_by: "name:asc",
    filter_by: `coords:[(32.7767, 96.7970, 50 mi),(34.0522, 118.2437, 20 mi)]`,
  }

  const results = await client.collections('offices').documents().search(q);
```

## Expected Behavior

Return documents that meet 1 of the geopoint requirements.

## Actual Behavior

Only 1 geopoint supported

## Metadata

**Typesense Version**: 24.x

**OS**: Linux

**Comment by kishorenc** [comment]
Can you try with the `||` or operator: 

`coords: (x1, y1, 20 mi) || cords:  (x2, y2,  20mi)`

**Comment by becauseinterwebs** [comment]
> Can you try with the `||` or operator:
> 
> `coords: (x1, y1, 20 mi) || cords: (x2, y2, 20mi)`

I tried:

```
const q = {
    q: "*",
    query_by: "email",
    filter_by: `coords:(32.7767, 96.7970, 50 mi) || coords:(34.0522, 118.2437, 20 mi)`,
  }
```
  
  and I get:
  
`  RequestMalformed: Request failed with HTTP code 400 | Server said: Unit must be either `km` or `mi`.`

**Comment by kishorenc** [comment]
@becauseinterwebs 

May I know what client you are using? It works for me via `curl`. See here: https://gist.github.com/kishorenc/a31df9c634303f9313de6d1022368c1c

Can you give me a similar fully reproducible snippet?

**Comment by becauseinterwebs** [comment]
The client is **1.5.3** (using NodeJs 18) and our server version is **0.23.1**.  Our instance is hosted by typesense.  When I try the sample query in the browser on our hosted instance:

`https://{our server instance}/collections/offices/search?q=*&filter_by=coords:(32.7767, 96.7970, 50 mi) || coords:(34.0522, 118.2437, 20 mi)&x-typesense-api-key=xxxxxxx`

I get:

`{ "message": "Not Found"}`

I also tried it in Postman and same response.

I just put in a request to have our server upgraded to **0.24.0**

I also noticed when I login to our cluster and try the search page that the **coords** field, although listed in the schema as a geopoint that is indexed and sortable, does not appear in the filter options in the search form on the page.  

I took the script you provided and added our typesense server and key.  It created the collection and added the records, but when I tried the search in my browser I got:

`{"message": "Unit must be either `km` or `mi`."}`

I'm hoping the upgrade to **0.24.0** will solve it.
